### ISSUE TITLE
Android Soft-Keyboard issue

### ISSUE THREAD

**Issue by NathanaelA** [NativeScript/NativeScript#783]
When I try to dismiss the soft keyboard on a page.ShowModal page; it throws this error:

From Console:
------------------
W/Binder  (  744): Caught a RuntimeException from the binder stub implementation.
W/Binder  (  744): java.lang.NullPointerException
W/Binder  (  744):      at android.inputmethodservice.IInputMethodWrapper.setSessionEnabled(IInputMethodWrapper.java:280)
W/Binder  (  744):      at com.android.internal.view.IInputMethod$Stub.onTransact(IInputMethod.java: 129)
W/Binder  (  744):      at android.os.Binder.execTransact(Binder.java:404)
W/Binder  (  744):      at dalvik.system.NativeStart.run(Native Method)
**W/InputMethodManagerService(  605): Got RemoteException sending setActive(false) notification to pid 1935 uid 10062**

---

pid 1935 was my process...

**Comment by NathanaelA** [comment]
I have only been able to duplicate this in Geny Image 4.4.4; it MIGHT be a bug in Geny however none of the other Apps on my geny emulator crash when the keyboard is being dismissed.  So not 100% who is at fault...

**Comment by N3ll** [comment]
@NathanaelA  - this does not look like a problem that is caused by {N}. Maybe you can try to put the dismiss method in a timeout and check whether there are some transactions to be handled before the dismissing of the soft keyboard.

**Comment by NathanaelA** [comment]
@N3ll - The dismiss is either by me clicking out of the field or just clicking the close on the soft keyboard.  Not me actually running any code to close the dialog.   

So this isn't something I can work around via a timeout.   That is why I stuck this might be a Geny issue; but since I haven't seen any other issues like this anywhere else using native apps; I have no idea who is to blame...   

Is their any code or callback called anywhere in the common code or runtimes that is called when the soft-keyboard is closed?

**Comment by N3ll** [comment]
@NathanaelA - we tried to reproduce the problem with this [demo project] (https://github.com/NativeScript/NativeScript/tree/master/apps/modal-views-demo). The version of Geny Motion that we used was 2.5.2 and the device Samsung Galaxy S4 - 4.4.4. Can you please send your code and information about the Geny version and which device you are testing on, so we can reproduce it exactly?

**Comment by NathanaelA** [comment]
I'm using Geny 2.4.0 (also on the Galaxy S4 - 4.4.4.) .   I'll download the demo project and see what happens on 2.4.0 -- maybe it is a Geny version issue, so I'll also pull down 2.5.4 and see if it fixes the issue..

**Comment by NathanaelA** [comment]
I'm not yet sure what is causing it.   I tried to do some diagnostics and I can't seem to duplicate it using the demo project, but my project I can fairly easily duplicate it on.   I'll keep plugging away and see if I can figure out what causes the issue.

**Comment by N3ll** [comment]
I can try to reproduce it if you send me your project and see what is different in yours and in the demo project.

**Comment by N3ll** [comment]
@NathanaelA - any luck with this one?

**Comment by NathanaelA** [comment]
@N3ll -- Sorry, I haven't had a chance since we last spoke to do anything more than verify it still occurs with v1.4 of the runtimes. The weird thing is that it only occurs with one of my test apps; not the others.  

So I'm not sure the cause yet.  I'll be making a repo for the test app soon as it is related to that other project that is on Google docs -- I just haven't had time to clean it up and remove all my "extra" test stuff from the source code since I want that repo to be as clean as possible for new developers.

 I have downloaded the new Geny, but I haven't installed it.   Too many projects going and not enough time.   

Don't close this issue; it is still occurring; just not sure why as I tried to take the same app that you use and make changes to make it have the same number of fields, opening dialogs, etc.  Hmm, it just dawned on me of one change; I do have a custom fonts in this project -- I'll have to see if rendering a custom font has something to do with the issue...
